v2/api/subscription/customizations/alias_extensions.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package customizations
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/subscription/armsubscription"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	storage "github.com/Azure/azure-service-operator/v2/api/subscription/v1beta20211001storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+var _ extensions.Deleter = &AliasExtension{}
+
+func (extension *AliasExtension) Delete(ctx context.Context, resolver *resolver.Resolver, armClient *genericarmclient.GenericClient, obj genruntime.ARMMetaObject, next extensions.DeleteFunc) (ctrl.Result, error) {
+	// First cancel the subscription, then delete the alias
+	typedObj, ok := obj.(*storage.Alias)
+	if !ok {
+		return ctrl.Result{}, errors.Errorf("cannot run on unknown resource type %T, expected *subscription.Alias", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not been updated to match
+	var _ conversion.Hub = typedObj
+
+	// Get the subscription ID
+	subscriptionID, ok := getSubscriptionID(typedObj)
+	if !ok {
+		// SubscriptionID isn't populated, allow deletion to proceed
+		return next(ctx, resolver, armClient, obj)
+	}
+
+	// Using armClient.ClientOptions() here ensures we share the same HTTP connection, so this is not opening a new
+	// connection each time through
+	subscriptionClient, err := armsubscription.NewClient(armClient.Creds(), armClient.ClientOptions())
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to create new workspaceClient")
+	}
+
+	// Don't need to do anything with the response here so just ignore it
+	_, err = subscriptionClient.Cancel(ctx, subscriptionID, nil)
```

Comment:
Might the cancellation be a long-running-operation that needs monitoring?

Reply:
It's not, this finishes synchronously. Which is good too because otherwise we'd have to hack this up further.

Reply:
I've added a note about this in the code. Also worth clarifying that when I said it finishes synchronously I mean that the SDK also doesn't do any polling here. There is no azure async operation to poll and this operation completes quickly as the result of a single POST response.